### ISSUE TITLE
doesntHave method does not work

### ISSUE THREAD

**Issue by joseraul** [ulobby/NeoEloquent#36]
- Laravel Version: 7.x
- PHP Version: 7.4
- NeoEloquent Version: 7.0.0

### Description:
The `doesntHave` method does not work. I guess other variants like `whereDoesntHave`, `orWhereDoesntHave` will not work either.

### Steps To Reproduce:

```
Person::doesntHave("twitter")
```
is generating a query like
```
MATCH (person:`Person`), (person)-[`rel_owns_twitter`:`OWNS`]->(twitter:`Twitter`) WHERE person.deleted_at is null WITH person, count(twitter) AS twitter_count WHERE twitter_count < 1 RETURN person
```

It is matching on the Persons Twitter account and then requiring that there is less than one Twitter account...

**Comment by raavus-funkmaster** [comment]
So `doesntHave` appears to call `has` in QueriesRelationships.php. I think if the second match clause was changed to an `OPTIONAL MATCH`, it would fix `doesntHave` without breaking `has`.

**Comment by xumng** [comment]
> So `doesntHave` appears to call `has` in QueriesRelationships.php. I think if the second match clause was changed to an `OPTIONAL MATCH`, it would fix `doesntHave` without breaking `has`.

yes. When I modify it, it can work. But where sub-clause don’t work. Because where sub-clause attach in optional match sub-clause not attach in match sub-clause.
